src/Shared/Model/Providers/ProviderFactory.cs

```diff
@@ -0,0 +1,68 @@
+// Copyright (c) Microsoft Corporation. Licensed under the MIT License.
+
+namespace Microsoft.CST.OpenSource.Model.Providers;
+
+using Contracts;
+using PackageUrl;
+using System;
+using System.Collections.Generic;
+using System.Linq;
+using System.Net.Http;
+
+public class ProviderFactory : IManagerProviderFactory
+{
+    /// <inheritdoc />
+    public IHttpClientFactory HttpClientFactory { get; }
+
+    public ProviderFactory(IHttpClientFactory? httpClientFactory = null)
+    {
+        HttpClientFactory = httpClientFactory ?? new DefaultHttpClientFactory();
+    }
+
+    /// <summary>
+    /// Creates a <see cref="BaseProvider"/>.
+    /// </summary>
+    /// <param name="httpClientFactory">The <see cref="IHttpClientFactory"/> to use in the <see cref="BaseProvider"/>. Defaults to null.</param>
+    /// <returns>A new <see cref="BaseProvider"/>.</returns>
+    public static BaseProvider CreateBaseProvider(IHttpClientFactory? httpClientFactory = null)
+    {
+        return new BaseProvider(httpClientFactory);
+    }
+
+    /// <summary>
+    /// Get an appropriate implementation of <see cref="BaseProvider"/> for the project manager in the provided
+    /// <see cref="PackageURL"/>. If no implementation of <see cref="BaseProvider"/> exists for this manager, returns null.
+    /// </summary>
+    /// <param name="purl">The <see cref="PackageURL"/> for the package to create the provider for.</param>
+    /// <returns>An implementation of <see cref="BaseProvider"/> for this project manager, or null.</returns>
+    public BaseProvider CreateProvider(PackageURL purl)
+    {
+        if (ManagerProviders.Count == 0)
+        {
+            ManagerProviders.AddRange(typeof(BaseProvider).Assembly.GetTypes()
```

Comment:
I am not a fan of this implicit enumeration - I think it would make sense to have the list of available providers set before you call this method - it seems more appropriate for the constructor - perhaps also as an optional parameter to skip this reflection.